engine/table/src/main/java/io/deephaven/engine/table/impl/by/GroupByChunkedOperator.java

```diff
@@ -494,67 +478,69 @@ public void propagateInitialState(@NotNull final QueryTable resultTable) {
+        try (final WritableRowSet stepDestinations = stepUpdatedDestinationBuilder.build()) {
+            // use the builders to modify the rowsets
+            if (stepDestinations.isNonempty()) {
+                try (final ResettableWritableObjectChunk<WritableRowSet, Values> rowSetResettableChunk =
+                             ResettableWritableObjectChunk.makeResettableChunk();
+                     final ResettableWritableObjectChunk<RowSetBuilderRandom, Values> addedBuildersResettableChunk =
+                             ResettableWritableObjectChunk.makeResettableChunk();
+                     final ResettableWritableObjectChunk<RowSetBuilderRandom, Values> removedBuildersResettableChunk =
+                             ResettableWritableObjectChunk.makeResettableChunk();
+                     final RowSequence.Iterator destinationsIterator =
+                             stepDestinations.getRowSequenceIterator()) {
+
+                    // noinspection unchecked
+                    final WritableObjectChunk<WritableRowSet, Values> rowSetBackingChunk =
+                            rowSetResettableChunk.asWritableObjectChunk();
+                    // noinspection unchecked
+                    final WritableObjectChunk<RowSetBuilderRandom, Values> addedBuildersBackingChunk =
+                            addedBuildersResettableChunk.asWritableObjectChunk();
+                    // noinspection unchecked
+                    final WritableObjectChunk<RowSetBuilderRandom, Values> removedBuildersBackingChunk =
+                            removedBuildersResettableChunk.asWritableObjectChunk();
+
+                    while (destinationsIterator.hasMore()) {
+                        final long firstSliceDestination = destinationsIterator.peekNextKey();
+                        final long firstBackingChunkDestination =
+                                rowSets.resetWritableChunkToBackingStore(rowSetResettableChunk, firstSliceDestination);
+                        addedBuilders.resetWritableChunkToBackingStore(addedBuildersResettableChunk, firstSliceDestination);
+                        removedBuilders.resetWritableChunkToBackingStore(removedBuildersResettableChunk, firstSliceDestination);
+
+                        final long lastBackingChunkDestination =
+                                firstBackingChunkDestination + rowSetBackingChunk.size() - 1;
+                        final RowSequence initialDestinationsSlice =
+                                destinationsIterator.getNextRowSequenceThrough(lastBackingChunkDestination);
+
+                        initialDestinationsSlice.forAllRowKeys((final long destination) -> {
+                            final int backingChunkOffset =
+                                    Math.toIntExact(destination - firstBackingChunkDestination);
+                            WritableRowSet workingRowSet = rowSetBackingChunk.get(backingChunkOffset);
```

Comment:
`getUnsafe`

Reply:
I didn't think `getUnsafe()` was available on a `WritableObjectChunk`

Reply:
It's not, I was mistakenly thinking this was an array source. Thank you for your generosity in pretending I might be right.